drivers/iio/proximity/vcnl3020.c

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * vcnl3020.c - Support for Vishay VCNL3020 proximity sensor
+ *
+ * based on vcnl4000.c
+ */
+
+#include <linux/module.h>
+#include <linux/i2c.h>
+#include <linux/err.h>
+#include <linux/delay.h>
+
+#include <linux/iio/iio.h>
+#include <linux/iio/sysfs.h>
+#include <linux/platform_device.h>
+#include <linux/iio/proximity/vcnl3020.h>
+
+static const struct i2c_device_id vcnl3020_id[] = {
+	{ "vcnl3020", 0 },
+	{}
+};
+MODULE_DEVICE_TABLE(i2c, vcnl3020_id);
+
+static int32_t vcnl3020_init(struct vcnl3020_data *data)
+{
+	s32 rc;
+	u32 proximity_rate, led_current, threshold, count_exceed, param_val;
+
+	struct device *dev = &data->client->dev;
+
+	rc = device_property_read_u32(dev, "proximity-rate", &param_val);
+	if (rc) {
+		dev_err(dev, "couldn't get proximity rate %x", rc);
+		goto exit;
+	}
+	proximity_rate = param_val;
+
+	rc = device_property_read_u32(dev, "led-current", &param_val);
+	if (rc) {
+		dev_err(dev, "Couldn't get led current %x", rc);
+		goto exit;
+	}
+	led_current = param_val;
+
+	rc = device_property_read_u32(dev, "threshold", &param_val);
+	if (rc) {
+		dev_err(dev, "Couldn't get threshold %x", rc);
+		goto exit;
+	}
+	threshold = param_val;
+
+	rc = device_property_read_u32(dev, "count-exceed", &param_val);
+	if (rc) {
+		dev_err(dev, "couldn't get count exceed %x", rc);
+		goto exit;
+	}
+	count_exceed = param_val;
+
+	rc = i2c_smbus_read_byte_data(data->client, VCNL_PROD_REV);
+	if (rc < 0) {
+		dev_err(dev, "Error in prod rev reading out %x", rc);
```

Comment:
`"Error (%d) reading product revision" `

The `rc` is usually a negated `errno` value, there is no need to print it in hex.